src/components/CreateGroupModal.vue

```diff
@@ -13,30 +13,150 @@
       <ion-item>
-        <ion-textarea :label="translate('Description')" labelPlacement="floating" />
+        <ion-textarea :label="translate('Description')" v-model="formData.description" labelPlacement="floating" />
+      </ion-item>
+      <ion-item>
+        <ion-label>{{ translate("Product store") }}</ion-label>
+        <ion-label slot="end">{{ eComStore.storeName ? eComStore.storeName : eComStore.productStoreId }}</ion-label>
+      </ion-item>
+      <ion-item>
+        <ion-select :label="translate('Group level configurations')" v-model="selectedConfigFacilityId" interface="popover">
+          <ion-select-option value="new">{{ translate("Create new") }}</ion-select-option>
+          <ion-select-option v-for="facility in configFacilities" :key="facility.facilityId" :value="facility.facilityId" >{{ facility.facilityName }}</ion-select-option>
+        </ion-select>
       </ion-item>
     </ion-list>
   </ion-content>
 
   <ion-fab vertical="bottom" horizontal="end" slot="fixed">
-    <ion-fab-button>
+    <ion-fab-button @click="createGroup()">
       <ion-icon :icon="checkmarkDone" />
     </ion-fab-button>
   </ion-fab>
 </template>
 
 <script setup lang="ts">
-import { IonButton, IonButtons, IonContent, IonFab, IonFabButton, IonHeader, IonIcon, IonInput, IonItem, IonList, IonTextarea, IonTitle, IonToolbar, modalController } from "@ionic/vue";
+import { IonButton, IonButtons, IonContent, IonFab, IonFabButton, IonHeader, IonIcon, IonInput, IonItem, IonLabel, IonList, IonSelect, IonSelectOption, IonText, IonTextarea, IonTitle, IonToolbar, modalController } from "@ionic/vue";
 import { closeOutline, checkmarkDone } from "ionicons/icons";
 import { translate } from "@/i18n";
+import { useStore } from "vuex";
+import { computed, ref } from "vue";
+import { generateInternalId, hasError, showToast } from "@/utils";
+import logger from "@/logger";
+import { ChannelService } from "@/services/ChannelService";
+
+const store = useStore();
+
+const formData = ref({
+  facilityGroupName: "",
+  facilityGroupId: "",
+  description: "",
+}) as any;
+const selectedConfigFacilityId = ref("new");
+const facilityGroupId = ref("") as any;
+
+const eComStore = computed(() => store.getters["user/getCurrentEComStore"])
+const configFacilities = computed(() => store.getters["util/getConfigFacilities"])
 
 function closeModal() {
   modalController.dismiss();
 }
+
+async function createGroup() {
+  if (!formData.value.facilityGroupName?.trim()) {
+    showToast(translate("Please fill in all the required fields"))
+    return;
+  }
+
+  if (formData.value.facilityGroupId.length > 20) {
+    showToast(translate("Internal ID cannot be more than 20 characters."))
+    return
+  }
+
+  // In case the user does not lose focus from the facility name input
+  // and click on create the button, we need to set the internal id manually
+  if (!formData.value.facilityGroupId) {
+    formData.value.facilityGroupId = generateInternalId(formData.value.facilityGroupName)
```

Comment:
If user enters a facilityGroupName having more than 20 characters then facilityGroupId generated in this step will be more than 20 characters, but we have previously added a check to not create facilityGroup if Id is more than 20 characters.